scripts/createlegends.js

```diff
@@ -39,16 +39,18 @@ let getStylesByResolution = function(data, styleType) {
   for (let i=0; i < data[styleType].length; i++) {
     let resolutionHash = 'res';
     let style = data[styleType][i];
-    if (style.minResolution) {
-      resolutionHash += '-min-' + style.minResolution.toString();
-    }
-    if (style.maxResolution) {
-      resolutionHash += '-max-' + style.maxResolution.toString();
-    }
-    if (!styles.hasOwnProperty(resolutionHash)) {
-      styles[resolutionHash] = [];
+    if (String(style.vectorOptions.src).indexOf('nodata') == -1) {
```

Comment:
It's cleaner to use RegExp to compare string content and you can't be sure vectorOptions exists so do:
```
if (style.vectorOptions && /nodata/.test(style.vectorOptions.src))
```
Anyway it seems really hackish,  the function stores styles by resolutions, I don't think the url of the image is a good reason to not store a styles in this array

Can you ask meteo guys why they don't want the no data image in there legend. It seems better idea to add it to the legend.

Reply:
@oterral 
1. thanks for the RegExp
2. the hack is, that the all nodata images are called nodataxx.png
3. the script is a oneshot in the context of generating meteoschweiz legends - based on a spreadseet taking the order of the entries into account...